Harvest.Web/Startup.cs

```diff
@@ -86,17 +86,33 @@ public void ConfigureServices(IServiceCollection services)
                 };
             });
 
+            // define accessCode => roleCode[] mapping once for both AccessHandler and UserService
+            static string[] GetRoles(string accessCode)
+            {
+                return accessCode switch
+                {
+                    // System can access anything
+                    AccessCodes.SystemAccess => new[] { Role.Codes.System },
+                    // FieldManager can access anything restricted to FieldManager, Supervisor, Worker or PI roles
+                    AccessCodes.FieldManagerAccess => new[] { Role.Codes.FieldManager },
+                    // Supervisor can access anything restricted to Supervisor, Worker or PI roles
+                    AccessCodes.SupervisorAccess => new[] { Role.Codes.Supervisor, Role.Codes.FieldManager },
+                    // Worker can access anything restricted to Worker role
+                    AccessCodes.WorkerAccess => new[] { Role.Codes.Worker, Role.Codes.Supervisor, Role.Codes.FieldManager },
+                    // PI can access anything restricted to PI role
+                    AccessCodes.PrincipalInvestigator => new[] { Role.Codes.PI, Role.Codes.Supervisor, Role.Codes.FieldManager },
+                    _ => throw new ArgumentException($"{nameof(accessCode)} is not a valid {nameof(AccessCodes)} constant")
+                };
+            }
+

```

Comment:
This is pretty nice.  throw it in a helper file somewhere though so we can keep startup slimmer.

Reply:
done